dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/h2/Http2ServerChannelObserver.java

```diff
@@ -74,12 +76,33 @@
 
     @Override
     public void cancel(Throwable throwable) {
+        if (throwable instanceof CancelStreamException) {
+            if (((CancelStreamException) throwable).isCancelByRemote()) {
+                closed = true;
+            }
+        }
```

Comment:
**Suggestion**: Refactor the condition inside the cancel method to reduce nesting and improve readability. <br> **Code Suggestion**: 
 ```
 +if (throwable instanceof CancelStreamException && ((CancelStreamException) throwable).isCancelByRemote()) {
 -    if (((CancelStreamException) throwable).isCancelByRemote()) {
 -        closed = true;
 +    closed.set(true);
 -    }
 ```